src/Store/ExtensionClientStoreConnector.re

```diff
@@ -161,6 +162,34 @@ module ExtensionDocumentHighlightProvider = {
   };
 };
 
+module ExtensionDocumentSymbolProvider = {
+  let create =
+      (
+        client,
+        documentSymbolProvider: Protocol.DocumentSymbolProvider.t,
+        buffer,
+      ) => {
+    Core.Buffer.getFileType(buffer)
```

Comment:
This pattern is common across all the extension host providers - extract to a helper function